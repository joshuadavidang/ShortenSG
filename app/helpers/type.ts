type KeydownEvent = React.KeyboardEventHandler<
  HTMLInputElement | HTMLTextAreaElement
>;

export type { KeydownEvent };
