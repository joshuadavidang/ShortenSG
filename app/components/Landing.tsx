import Form from "./Form";
import Image from "next/image";
import LinkShortenHero from "~/undraw_link_shortener.png";

export default function Landing() {
  return (
    <div className="flex flex-col md:flex-row w-full lg:gap-32 justify-center items-center md:relative lg:bottom-28">
      <Image
        src={LinkShortenHero}
        width={500}
        alt="Picture of the author"
        priority
      />
      <div className="flex flex-col px-8 md:w-1/2">
        <div className="mb-6">
          <h1 className="font-bold mb-3">Bid farewell to long and ugly URLs</h1>
          <h2 className="font-semibold mb-3">✂️ Shorten them now</h2>
          <p className="text-gray-500">
            One step closer to looking credible ⤵
          </p>
        </div>
        <Form />
      </div>
    </div>
  );
}
