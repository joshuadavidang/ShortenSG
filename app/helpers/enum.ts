enum DATA {
  AVAILABLE = "Available",
  UNAVAILABLE = "Unavaialble",
}

export { DATA };
