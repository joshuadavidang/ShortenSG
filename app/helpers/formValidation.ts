function validateFormLength(text: string) {
  return text.length > 0;
}

export { validateFormLength };
