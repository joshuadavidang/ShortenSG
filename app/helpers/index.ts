export * from "./copy";
export * from "./formValidation";
export * from "./smoothScrollTo";
export * from "./type";
