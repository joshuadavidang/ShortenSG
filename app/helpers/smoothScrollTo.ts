const smoothScrollTo = (id: string) => {
  const element = document.getElementById(id);
  if (element) {
    window.scrollTo({
      top: element.getBoundingClientRect().top + window.scrollY,
      left: 0,
      behavior: "smooth",
    });
  }
};

export { smoothScrollTo };
