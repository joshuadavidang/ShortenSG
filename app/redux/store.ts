"use client";

import { configureStore } from "@reduxjs/toolkit";
import storage from "redux-persist/lib/storage";
import {
  FLUSH,
  REGISTER,
  REHYDRATE,
  PAUSE,
  persistReducer,
  persistStore,
  PERSIST,
  PURGE,
} from "redux-persist";
import urlReducer from "./features/url/urlSlice";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, urlReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const persistor = persistStore(store);
