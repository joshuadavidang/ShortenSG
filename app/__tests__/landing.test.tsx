import Landing from "@/components/Landing";
import { render, screen } from "@testing-library/react";
import { Providers } from "@/redux/provider";
import "@/helpers/matchMedia";
import "@testing-library/jest-dom";

beforeEach(() => {
  jest.spyOn(console, "error");
  /* eslint-disable no-console */
  // @ts-expect-error jest.spyOn adds this functionallity
  console.error.mockImplementation(() => null);
});

describe("Landing Page", () => {
  test("Should renders and matches the user interface correctly", () => {
    const { asFragment } = render(
      <Providers>
        <Landing />
      </Providers>,
    );
    const fragment = asFragment();
    expect(fragment).toMatchSnapshot();
  });

  test("Should render 3 headings in the landing page", () => {
    render(
      <Providers>
        <Landing />
      </Providers>,
    );
    expect(
      screen.getByText("Bid farewell to long and ugly URLs"),
    ).toBeInTheDocument();
    expect(screen.getByText("✂️ Shorten them now")).toBeInTheDocument();
    expect(
      screen.getByText("One step closer to looking credible ⤵"),
    ).toBeInTheDocument();
  });
});
