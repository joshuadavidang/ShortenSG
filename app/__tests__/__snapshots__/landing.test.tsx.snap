// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Should renders and matches the user interface correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col md:flex-row w-full lg:gap-32 justify-center items-center md:relative lg:bottom-28"
  >
    <img
      alt="Picture of the author"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="500"
      src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
      style="color: transparent;"
      width="500"
    />
    <div
      class="flex flex-col px-8 md:w-1/2"
    >
      <div
        class="mb-6"
      >
        <h1
          class="font-bold mb-3"
        >
          Bid farewell to long and ugly URLs
        </h1>
        <h2
          class="font-semibold mb-3"
        >
          ✂️ Shorten them now
        </h2>
        <p
          class="text-gray-500"
        >
          One step closer to getting neat & shareable URL ⤵
        </p>
      </div>
      <div
        class="lg:w-10/12 md:p-16 md:bg-blue rounded-2xl"
      >
        <form
          class="flex flex-col gap-3"
        >
          <div
            class="flex justify-center items-center gap-3 mb-3"
          >
            <svg
              class="w-6 h-auto opacity-75"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.51194 3.00541C9.18829 2.54594 10.0435 2.53694 10.6788 2.95419C10.8231 3.04893 10.9771 3.1993 11.389 3.61119C11.8009 4.02307 11.9513 4.17714 12.046 4.32141C12.4633 4.95675 12.4543 5.81192 11.9948 6.48827C11.8899 6.64264 11.7276 6.80811 11.3006 7.23511L10.6819 7.85383C10.4867 8.04909 10.4867 8.36567 10.6819 8.56093C10.8772 8.7562 11.1938 8.7562 11.389 8.56093L12.0077 7.94221L12.0507 7.89929C12.4203 7.52976 12.6568 7.2933 12.822 7.0502C13.4972 6.05623 13.5321 4.76252 12.8819 3.77248C12.7233 3.53102 12.4922 3.30001 12.1408 2.94871L12.0961 2.90408L12.0515 2.85942C11.7002 2.508 11.4692 2.27689 11.2277 2.11832C10.2377 1.46813 8.94398 1.50299 7.95001 2.17822C7.70691 2.34336 7.47044 2.57991 7.1009 2.94955L7.058 2.99247L6.43928 3.61119C6.24401 3.80645 6.24401 4.12303 6.43928 4.31829C6.63454 4.51355 6.95112 4.51355 7.14638 4.31829L7.7651 3.69957C8.1921 3.27257 8.35757 3.11027 8.51194 3.00541ZM4.31796 7.14672C4.51322 6.95146 4.51322 6.63487 4.31796 6.43961C4.12269 6.24435 3.80611 6.24435 3.61085 6.43961L2.99213 7.05833L2.94922 7.10124C2.57957 7.47077 2.34303 7.70724 2.17788 7.95035C1.50265 8.94432 1.4678 10.238 2.11799 11.2281C2.27656 11.4695 2.50766 11.7005 2.8591 12.0518L2.90374 12.0965L2.94837 12.1411C3.29967 12.4925 3.53068 12.7237 3.77214 12.8822C4.76219 13.5324 6.05589 13.4976 7.04986 12.8223C7.29296 12.6572 7.52943 12.4206 7.89896 12.051L7.89897 12.051L7.94188 12.0081L8.5606 11.3894C8.75586 11.1941 8.75586 10.8775 8.5606 10.6823C8.36533 10.487 8.04875 10.487 7.85349 10.6823L7.23477 11.301C6.80777 11.728 6.6423 11.8903 6.48794 11.9951C5.81158 12.4546 4.95642 12.4636 4.32107 12.0464C4.17681 11.9516 4.02274 11.8012 3.61085 11.3894C3.19896 10.9775 3.0486 10.8234 2.95385 10.6791C2.53661 10.0438 2.54561 9.18863 3.00507 8.51227C3.10993 8.35791 3.27224 8.19244 3.69924 7.76544L4.31796 7.14672ZM9.62172 6.08558C9.81698 5.89032 9.81698 5.57373 9.62172 5.37847C9.42646 5.18321 9.10988 5.18321 8.91461 5.37847L5.37908 8.91401C5.18382 9.10927 5.18382 9.42585 5.37908 9.62111C5.57434 9.81637 5.89092 9.81637 6.08619 9.62111L9.62172 6.08558Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            <input
              class="flex h-12 w-full rounded-md bg-gray-100 px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50"
              name="originalURL"
              placeholder="Paste your long url here"
              type="text"
            />
          </div>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-12 rounded-md px-8 w-full"
            type="submit"
          >
            Shorten URL
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
